### PR TITLE
Add unit tests for deposit, withdraw, increase amount and increase unlock time

### PR DESCRIPTION
**Pull request by nightowl121**

### CODE REVIEW

File: token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        for (Account account : accounts) {
+            balance = (BigInteger) bBalnScore.call("balanceOfAt", account.getAddress(),
+                    BigInteger.valueOf(sm.getBlock().getHeight() - 4));
+            totalSupplyAtIncreasedBlock = totalSupplyAtIncreasedBlock.add(balance);
+        }
+        assertEquals(bBalnScore.call("totalSupplyAt", BigInteger.valueOf(sm.getBlock().getHeight() - 4)),
+                totalSupplyAtIncreasedBlock);
+    }
+
+}
```
Comment (AntonAndell) [inline token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java:440]: Many tests are very hard to read, my above comments are some examples of how you can improve the readability of your code. They are not needed but to have rapid development in the future it is a very good practice to have.

**Comment by nightowl121** [comment]
Thanks @AntonAndell for the suggestions, these are really helpful. Will work on making the code more readable.

File: token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java [hunk 2]
```diff
@@ -72,10 +72,13 @@ public VotingBalance() {
 
     @BeforeEach
     public void setup() throws Exception {
+        //Deploy token contract and boosted baln contract
         tokenScore = sm.deploy(owner, BalnToken.class, "Balance Token", "BALN", 18);
         bBalnScore = sm.deploy(owner, BoostedBaln.class, tokenScore.getAddress(), "Boosted Baln", "bBALN");
 
-        for (int counter = 0; counter < 10; counter++) {
+        //Create 10 accounts
```
Comment (AntonAndell) [inline token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java:79]: No need for this comment, the code explains it instead

File: token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java [hunk 3]
```diff
@@ -121,7 +128,7 @@ void createLockTest() {
         @Test
         void unlockTimeLessThanCurrentTime() {
             // Create lock in past
```
Comment (AntonAndell) [inline token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java:130]: This one is not needed either

File: token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java [hunk 4]
```diff
@@ -134,7 +141,7 @@ void unlockTimeLessThanCurrentTime() {
         void lockGreaterThanMaxTime() {
             // Create lock with greater than max time
             // 208 weeks is the maximum someone can lock tokens
```
Comment (AntonAndell) [inline token-contracts/bBaln/src/test/java/network/balanced/score/tokens/StateMachineTest.java:143]: No longer needed either.
Also The max variable is 209 while max lock is 208, mabye change is to 208 and use MAXIMUM_LOCK_WEEKS + 1 so we keep the numbers correct.

**Review by AntonAndell — COMMENTED** [review]
A lot of nice improvements, more could be done but I don't have any specific knowledge on what code standard we want in the balanced code base.
You can wait for @redlarva to review also before you accompany any more of my suggestions if u want. And i can help to review more if he think it is necessary.

**Review by redlarva — APPROVED** [review]

**Pull request merged**